py/runtime.c

```diff
@@ -1339,19 +1339,26 @@ mp_vm_return_kind_t mp_resume(mp_obj_t self_in, mp_obj_t send_value, mp_obj_t th
 
 mp_obj_t mp_make_raise_obj(mp_obj_t o) {
     DEBUG_printf("raise %p\n", o);
-    if (mp_obj_is_exception_type(o)) {
+    if (mp_obj_is_exception_instance(o)) {
```

Comment:
I suppose it might be more efficient to keep the `mp_obj_is_exception_type` check first, but instead of returning immediately, assign to `o`. Then reinstate the `mp_obj_is_exception_instance` check, but behind `if` instead of `else if`.